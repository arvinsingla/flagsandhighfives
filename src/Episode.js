import React, { Component } from 'react';

class Episode extends Component {

  playEpisode = (episodeId, autoplay = false) => {
    this.props.play(episodeId, autoplay);
  }

  render() {
    const { id, title, created_at, description } = this.props.data;

    return (
      <article className="content__item" key={id}>
        <header className="content__item__header">
          <button className="content__item__header__play" onClick={() => this.playEpisode(id, true)}>Play</button>
          <div className="content__item__header__text">
            <h3 className="content__item__header__text__title">{title}</h3>
            <time className="content__item__header__text__date">Published: {created_at}</time>
          </div>
        </header>
        <section className="content__item__description">{description}</section>
      </article>
    );
  }
}

export default Episode;
