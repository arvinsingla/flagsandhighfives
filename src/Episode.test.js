import React from 'react';
import Episode from './Episode';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

it('renders correctly', () => {
  const episode = {
    id: '123456',
    title: 'Hello World',
    created_at: '06-06-1981',
    description: 'I am a very model of a modern major general'
  }
  const tree = renderer.create(
    <Episode data={episode} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('calls callback on episode play', () => {
  const episode = {
    id: '123456',
    title: 'Hello World',
    created_at: '06-06-1981',
    description: 'I am a very model of a modern major general'
  }
  const play = (id, autoplay) => {
    expect(id).toBe(episode.id);
  }
  const episodeRender = shallow(
    <Episode data={episode} play={play} />
  );
  episodeRender.find('.content__item__header__play').simulate('click');
});
