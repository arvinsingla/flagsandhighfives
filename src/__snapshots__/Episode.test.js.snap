exports[`test renders correctly 1`] = `
<article
  className="content__item">
  <header
    className="content__item__header">
    <button
      className="content__item__header__play"
      onClick={[Function]}>
      Play
    </button>
    <div
      className="content__item__header__text">
      <h3
        className="content__item__header__text__title">
        Hello World
      </h3>
      <time
        className="content__item__header__text__date">
        Published: 
        06-06-1981
      </time>
    </div>
  </header>
  <section
    className="content__item__description">
    I am a very model of a modern major general
  </section>
</article>
`;
